agent/__init__.py

```diff
@@ -542,14 +542,17 @@ def clear_credentials(path):
                 credentials_grouped[owner][name] = {}
             credentials_grouped[owner][name][cred['key']] = cred['value']
 
+        root_pw = pwd.getpwnam("root")
+
         for owner in credentials_grouped:
 
-            pw = pwd.getpwnam("root")
+            pw = root_pw  # if no owner, use 'root'
             if owner:
                 try:
                     pw = pwd.getpwnam(owner)
-                except KeyError:
-                    pw = pwd.getpwnam("root")
+                except KeyError:  # if invalid owner, use 'root'
+                    print("Warning. There are credentials with wrong owner ({}). Skipped.".format(owner))
```

Comment:
That comment seems to be inconsistent with what was discussed in the comment.

Reply:
removed